src/savegame.cpp

```diff
@@ -999,12 +999,14 @@ void game::unserialize_master(std::istream &fin) {
        popup_nowait(_("Cannot find loader for save data in old version %d, attempting to load as current version %d."),savegame_loading_version, savegame_version);
     }
     try {
-        // single-pass parsing example
         JsonIn jsin(fin);
+        // single-pass parsing example
         jsin.start_object();
         while (!jsin.end_object()) {
             std::string name = jsin.get_member_name();
-            if (name == "next_mission_id") {
+            if( name == "item_uid" ) {
```

Comment:
The label probably shouldn't be written by game at all in that case, it should defer to the uid factory.

Yes, people play on shared servers, and it's a play mode we would like to support.  It does have issues with map data becoming inconsistent, but nothing like the code derailing like would happen if the uid counter were rolled backwards.